src/protocolsupport/protocol/typeremapper/itemstack/ItemStackRemapper.java

```diff
@@ -11,12 +11,11 @@ public static NetworkItemStack remapToClient(ProtocolVersion version, String loc
 		setComplexlyRemapped(itemstack, false);
 		itemstack = ItemStackComplexRemapperRegistry.remapToClient(version, locale, itemstack);
 		itemstack.setTypeId(LegacyItemType.REGISTRY.getTable(version).getRemap(itemstack.getTypeId()));
+
 		if (version == ProtocolVersion.MINECRAFT_PE) {
 			int peCombinedId = PEItems.getPECombinedIdByModernId(itemstack.getTypeId());
 			itemstack.setTypeId(PEItems.getIdFromPECombinedId(peCombinedId));
-			if (!isComplexlyRemapped(itemstack)) {
```

Comment:
I found something wrong with this too yeah. But we cannot just remove it, since it will mess up all our itemstack nbt remappers!

Reply:
hmm so i was curious about that. I was setting the typeID to the modern id and the data to 0 before- but the data defaults to 0, so it would have been always 'complex remapped'. but now i removed the setting data to 0 so i dont know :(

Reply:
should i just always run:

```
itemstack.setLegacyData(PEItems.getDataFromPECombinedId(peCombinedId));
```

here? its basically a new item stack, so i think its safe to assume it always needs to be run through the complex remapper.

Reply:
hmm so given that the PC -> PE conversion doesnt do any complex remapping, should be safe to always do this here?
```
setComplexlyRemapped(itemstack, false);
```

Reply:
hmm things get weird unless i set the data to 0 =\